pkgs/by-name/os/ossia-score/package.nix

```diff
@@ -0,0 +1,137 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, cmake
+, ninja
+, pkg-config
+, alsa-lib
+, avahi
+, bluez
+, boost185
+# , faust
+, fmt
+, ffmpeg
+, fftw
+, flac
+, git
+, gnutls
+, lame
+, libcoap
+, libjack2
+, libopus
+, libsamplerate
+, libsndfile
+, libvorbis
+, lilv
+, lv2
+, mpg123
+, pipewire
+, portaudio
+, portmidi
+, qt6
+, rapidfuzz-cpp
+, re2
+, SDL2
+, spdlog
+, suil
+, udev
+}:
+
+# TODO: figure out LLVM jit
+# assert lib.versionAtLeast llvm.version "15";
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ossia-score";
+  version = "3.2.0";
+
+  src = fetchFromGitHub {
+    owner = "ossia";
+    repo = "score";
+    rev = "fd8efedb29be05ce3b7ea067fdfdcae6537ebb47";
```

Comment:
This doesn't seem to be on the 3.2.0 tag: https://github.com/ossia/score/commit/ea4e943c6f7819824a24b4de5226b6c822debd06

If there are some upstream changed that we need, we can add them through patching, for example like this:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/waylandpp/default.nix#L28-L38

Reply:
any way to cherry-pick a specific commit instead of a patch file ? it's these two patches in the libossia submodule which are necessary :
https://github.com/ossia/libossia/commit/23f64e41473905d4b5c8c0267140dc60412253ee
https://github.com/ossia/libossia/commit/130e64f916c6805b7bb6bae5617f59051e968e6c

Reply:
Yes, like shown in the link, you can do something like this:

```nix
  patches = [
    # A nice comment explaining why this is needed
    (fetchpatch {
      name = "fix-missing-libsamplerate.patch";
      url = "https://github.com/ossia/libossia/commit/23f64e41473905d4b5c8c0267140dc60412253ee.patch";
      hash = ""; # to be filled
    })

    # Another comment
    (fetchpatch {
      # ...
    })
  ];
```

You will need to add `fetchpatch` in the arguments at the top.

Reply:
oh ! i didn't know this github feature at all

Reply:
hmm.. the problem is : the patch applies to a submodule, not to the main repo. I tried setting a path with: 

    extraPrefix = "/3rdparty/libossia";

in the fetchpatch invocation but to no avail.

Reply:
okay, found another way

Reply:
Can you try `extraPrefix = "3rdparty/libossia/";` instead? It seems to be what people are doing from a quick look inside nixpkgs